framework/web/Request.php

```diff
@@ -196,6 +268,39 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the [[trustedHosts]].
+     * @param HeaderCollection $headerCollection
+     * @since 2.0.13
+     */
+    protected function filterHeaders(HeaderCollection $headerCollection)
+    {
+        // do not trust any of the [[secureHeaders]] by default
+        $trustedHeaders = [];
+
+        // check if the client is a trusted host
+        if (!empty($this->trustedHosts)) {
+            $host = $this->getRemoteHost();
+            $ip = $this->getRemoteIP();
+            foreach ($this->trustedHosts as $hostRegex => $headers) {
+                if (!is_array($headers)) {
+                    $hostRegex = $headers;
+                    $headers = $this->secureHeaders;
+                }
+                if (preg_match($hostRegex, $host) || preg_match($hostRegex, $ip)) {
+                    $trustedHeaders = $headers;
```

Comment:
This will cause unexpected behavior when both host and ip are matched. Might not be an issue, but should at least be documented.

(Not something you introduced; just something I noticed just now)